### PR TITLE
Split zkApps account precondition failures

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR generates a distinct failure for each failed check in the zkApps account precondition. This is helpful for debugging, and should also allow users to identify issues when testing their transactions / zkApps.

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

File: src/lib/mina_base/transaction_status.ml [hunk 1]
```diff
@@ -39,7 +39,20 @@ module Failure = struct
         | Parties_replay_check_failed
         | Fee_payer_nonce_must_increase
         | Fee_payer_must_be_signed
-        | Account_precondition_unsatisfied
+        | Account_balance_precondition_unsatisfied
+        | Account_nonce_precondition_unsatisfied
+        | Account_receipt_chain_hash_precondition_unsatisfied
+        | Account_delegate_precondition_unsatisfied
+        | Account_sequence_state_precondition_unsatisfied
+        | Account_app_state_0_precondition_unsatisfied
```
Comment (psteckler) [inline src/lib/mina_base/transaction_status.ml:47]: Is it possible to have a single constructor that takes an integer argument?

Comment (psteckler) [reply src/lib/mina_base/transaction_status.ml:47]: I guess that doesn't work if you need `enum`

File: src/lib/transaction_snark/transaction_snark.ml [hunk 2]
```diff
@@ -1953,10 +1953,14 @@ module Base = struct
             (protocol_state_predicate, global_state) ->
             Zkapp_precondition.Protocol_state.Checked.check
               protocol_state_predicate global_state.protocol_state
-        | Check_account_precondition (_is_start, { party; _ }, account, _global)
-          ->
-            Zkapp_precondition.Account.Checked.check
-              party.data.account_precondition account.data
+        | Check_account_precondition ({ party; _ }, account, local_state) ->
+            let local_state = ref local_state in
```
Comment (psteckler) [inline src/lib/transaction_snark/transaction_snark.ml:1957]: it might be better to use a different name to distinguish it from the variable in the pattern

**Review by psteckler — APPROVED** [review]
couple of nits, looks fine otherwise

File: src/lib/mina_base/transaction_status.ml [hunk 3]
```diff
@@ -85,15 +91,35 @@ module Failure = struct
 
   let failure_max = max
 
-  let failure_num_bits =
-    let num_values = failure_max - failure_min + 1 in
-    Int.ceil_log2 num_values
+  let all =
```
Comment (deepthiskumar) [inline src/lib/mina_base/transaction_status.ml:94]: :bulb:

**Review by deepthiskumar — APPROVED** [review]

**Pull request merged**